### PR TITLE
Two small features: Auto-save and auto-reload

### PR DESCRIPTION
**Pull request by TimMensch**
I end up using a lot of editors when I'm developing, and so I like to have two features available:
1. Auto-save on loss of focus. I really, really don't want to have to remember to save everything before swapping away from one editor to another.

and
1. Auto-reload when an external file has been modified.

This patch adds both of these features as options (default disabled):

editor.saveonfocusloss = true
editor.autoreload = true

Tim

### CODE REVIEW

**Comment by madmaxoft** [comment]
@pkulchenko The trailing whitespace removal plugin should be somewhere more visible - perhaps mention it in the docs?

I spent half an hour looking for this feature, I was already deciding to file an issue. Now I can see the plugin is actually in the ZeroBranePackage repo.

**Comment by pkulchenko** [comment]
@madmaxoft, I'm open to suggestions on how to make it more visible. I've been thinking about listing some of the plugins on the Features page and maybe in Tips and Tricks section. The package manager is also coming at some point, which will allow to search for packages.

**Comment by d9k** [comment]
oh, I found striptrailingwhitespace plugin here, among the others: https://github.com/pkulchenko/ZeroBranePackage/blob/master/striptrailingwhitespace.lua
